lib/process-services-cloud/src/lib/form/services/content-cloud-node-selector.service.spec.ts

```diff
@@ -0,0 +1,124 @@
+import { ProcessServiceCloudTestingModule } from '../../testing/process-service-cloud.testing.module';
+import { TranslateModule } from '@ngx-translate/core';
+import { ContentCloudNodeSelectorService } from 'process-services-cloud';
+import { MatDialog, MatDialogModule } from '@angular/material/dialog';
+import { of, Subject } from 'rxjs';
+
+describe('ContentCloudNodeSelectorService', () => {
+    let service: ContentCloudNodeSelectorService;
+    let apiService: AlfrescoApiService;
+    let notificationService: NotificationService;
+    let getNodeSpy: jasmine.Spy;
+    let dialog: MatDialog;
+    let openDialogSpy: jasmine.Spy;
+    let showWarningSpy: jasmine.Spy;
+
+    const relativePathNodeResponseBody = {
+        entry: {
+            id: 'mock-relative-path-node-id'
+        }
+    };
+
+    const aliasNodeResponseBody = {
+        entry: {
+            id: 'mock-alias-node-id'
+        }
+    };
+
+    setupTestBed({
+        imports: [TranslateModule.forRoot(), ProcessServiceCloudTestingModule, MatDialogModule]
+    });
+
+    beforeEach(() => {
+        service = TestBed.inject(ContentCloudNodeSelectorService);
+        notificationService = TestBed.inject(NotificationService);
+        apiService = TestBed.inject(AlfrescoApiService);
+        dialog = TestBed.inject(MatDialog);
+
+        showWarningSpy = spyOn(notificationService, 'showWarning');
+        openDialogSpy = spyOn(dialog, 'open').and.returnValue({
+            afterOpened: () => of({}),
+            afterClosed: () => of({}),
+            componentInstance: {
+                body: '',
+                error: new Subject<any>()
+            }
+        });
+        getNodeSpy = spyOn(apiService.nodesApi, 'getNode');
+    });
+
+    it('should be able to set sourceNodeNotFound value to true if the relative path is invalid/deleted', async () => {
+        expect(service.sourceNodeNotFound).toBe(false);
+
+        getNodeSpy.and.returnValue(Promise.reject('Not exists'));
+        await service.fetchNodeIdFromRelativePath('mock-alias', { relativePath: 'mock-wrong-relativePath' });
+
+        expect(getNodeSpy).toHaveBeenCalledWith('mock-alias', {
+            relativePath: 'mock-wrong-relativePath'
+        });
+        expect(service.sourceNodeNotFound).toBe(true);
+    });
+
+    it('should be able to set sourceNodeNotFound value to false after the dialog close', async () => {
+        service.sourceNodeNotFound = true;
+        service.openUploadFileDialog('nodeId', 'single', true, true);
+
+        expect(openDialogSpy).toHaveBeenCalled();
+
+        service.close();
+
+        expect(service.sourceNodeNotFound).toBe(false);
+    });
+
+    it('should be able to show an notification if the relative path is invalid/deleted', () => {
+        service.sourceNodeNotFound = true;
+        service.openUploadFileDialog('nodeId', 'single', true, true);
+
+        expect(openDialogSpy).toHaveBeenCalled();
```

Comment:
same here. Imagine if someone refactors the openUploadFileDialog method. Why will they need to change a test that is called `should be able to show a notification` ?

Reply:
We are not showing notification before dialog opens, We show it after dialog opens. If someone changing/refactoring openUploadFileDialog they should change notification logic as well